src/main/java/randoop/test/RegressionCaptureGenerator.java

```diff
@@ -253,4 +277,44 @@ public RegressionChecks generateTestChecks(ExecutableSequence sequence) {
     }
     return checks;
   }
+
+  /**
+   * Returns true if the method:
+   *
+   * <ul>
+   *   <li>has one formal parameter
+   *   <li>has a return type of primitive, boxed primitive, String, or enum
+   *   <li>is side-effect-free
+   * </ul>
+   *
+   * @param tco side-effect-free method as a TypedClassOperation
+   * @return whether we can use this method as a side-effect-free assertion
+   */
+  public static boolean isAssertableSideEffectFree(TypedClassOperation tco) {
+    // When outputting checks, ignore side-effect-free methods that don't take a single
+    // argument.
+    if (tco.getInputTypes().size() != 1) {
+      return false;
+    }
+
+    if (tco.getOutputType().isVoid()) {
+      return false;
+    }
+
+    Class<?> outputClass = tco.getOutputType().getRuntimeClass();
+
+    if (outputClass == null) {
```

Comment:
How can this be true?  You already tested that the output type is not void.

Reply:
I was wondering about that as well - this if statement does actually catch a few cases (which is why I added it). From the documentation it looks like the outputType could be a NullReferenceType and calling getRuntimeClass() on it could return null while still not have isVoid() be true? I'm not sure how exactly it is generated though.

Reply:
OK, thanks for adding the comment to explain.